### PR TITLE
PoC of AIP-69 - A first task can be called remote

### PR DESCRIPTION
**Pull request by jscheffl**
This PR is a DRAFT to further discuss technical details on AIP-69 direction. DO NOT MERGE - but please feedback on it.

Please treat this as a PoC, it is only partly working. It can be used to distribute the first task but a lot of core required functions to make a MVP are missing. Status of implementation is tracked in `airflow/providers/remote/TODO.md` in this PR.

If you want to test it you can:
- Checkout the branch
- Run `breeze start-airflow --python 3.12 --load-example-dags --backend postgres --executor RemoteExecutor --answer y`
- Start the Remote executor with `airflow/providers/remote/_start_remote_worker.sh`
- A (non templated, not mapped, non failing) task can be run, e.g. use example DAG `example_python_operator`. Wohoo! 
image
- If you open the UI you will see an additional plugin loaded in the menu "Admin" -> "Remote Worker Status": 
image

Based on progress and time I will sometimes rebase and push changes to this branch. But it will not be something that will be review-able.
I will not add any migration logic in case of updates for the additional tables in DB Scheme. If I update the draft PR and DB schema changes, DB might need to be reset. Of course migration support would need to be added before a "real" release.

### CODE REVIEW

File: airflow/api_internal/internal_api_call.py
Comment (jscheffl) [inline airflow/api_internal/internal_api_call.py:1]: Note: This change is needed allowing to re-route the Internal API calls to another endpoint (Assuming: Remote Worker API should be HTTPS (not HTTP) and needs to be exposed Internet-Facing. (App GW, Protection different than internal API)

File: airflow/example_dags/example_subdag_operator.py
Comment (jscheffl) [inline airflow/example_dags/example_subdag_operator.py:1]: Note: I (temporarily) deleted this example as it makes compile issues with internal API enabled... maybe we should delete this example....

File: airflow/providers/remote/__init__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import packaging.version
+
+from airflow import __version__ as airflow_version
+
+__all__ = ["__version__"]
+
+__version__ = "0.1.0"
+
+if packaging.version.parse(packaging.version.parse(airflow_version).base_version) < packaging.version.parse(
+    "2.7.0"
```
Comment (jscheffl) [inline airflow/providers/remote/__init__.py:35]: Will need 2.10 at least!

File: airflow/providers/remote/_start_remote_worker.sh
Comment (jscheffl) [inline airflow/providers/remote/_start_remote_worker.sh:1]: This is a temporary script for development testing to start the remote worker.

Note that mainly it is used to drop all DB connection information to ensure that all backend calls must be made via HTTP API.

File: airflow/providers/remote/provider.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  Handle remote workers via HTTP(s) connection and distribute work over distributed sites
+
+state: not-ready
+source-date-epoch: 0
+# note that those versions are maintained by release manager - do not update them manually
+versions:
+  - 0.1.0
+
+dependencies:
+  - apache-airflow>=2.7.0
```
Comment (jscheffl) [inline airflow/providers/remote/provider.yaml:32]:
```suggestion
  - apache-airflow>=2.10.0
```

File: airflow/serialization/serialized_objects.py
Comment (jscheffl) [inline airflow/serialization/serialized_objects.py:1]: Note: This extension is needed to allow new Pydantic class mappings from Remote Provider Package to be registered - allowing to leverage AIP-44 RPC call logic with non-code model classes

File: docs/apache-airflow-providers-remote/celery_executor.rst
Comment (jscheffl) [inline docs/apache-airflow-providers-remote/celery_executor.rst:1]: Note: Don't take this RST serious - it is at the moment a copy&paste from celery as starting point/template.Does not make a benefit to review docs as of now.

File: airflow/providers/remote/plugins/templates/__init__.py
Comment (ashb) [inline airflow/providers/remote/plugins/templates/__init__.py:1]: Nit: this folder isn't a python module so we don't need this file 😁

Comment (jscheffl) [reply airflow/providers/remote/plugins/templates/__init__.py:1]: Yeah tried to but the pre-commit hook `check-providers-subpackages-init-file-exist` insists on it and always re-adds it :-D

Comment (potiuk) [reply airflow/providers/remote/plugins/templates/__init__.py:1]: You can always add exceptions for `templates` folder in the pre-commit. We already exclude `doc` and `adr` folders in it:

https://github.com/apache/airflow/blob/main/scripts/ci/pre_commit/check_providers_subpackages_all_have_init.py#L35

```python
ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir))


def check_dir_init_file(provider_files: list[str]) -> None:
    missing_init_dirs: list[Path] = []
    for candidate_path in provider_files:
        if candidate_path.endswith("/__pycache__"):
            continue
        path = Path(candidate_path)
        if path.is_dir() and not (path / "__init__.py").exists():
            if path.name != "adr" and path.name != "doc":
                missing_init_dirs.append(path)

```

**Comment by jscheffl** [comment]
Yeah, cool. Stale... but code "almost" is on main... closing but keeping as reference of progress.

**Pull request closed without merging**